packages/wrapped-keys/src/lib/api/sign-transaction-with-encrypted-key.ts

```diff
@@ -0,0 +1,38 @@
+import { signTransactionWithLitAction } from '../lit-actions-client';
+import { getLitActionCid } from '../lit-actions-client/utils';
+import { fetchPrivateKeyMetadata } from '../service-client';
+import { SignTransactionWithEncryptedKeyParams } from '../types';
+import { getFirstSessionSig, getPkpAccessControlCondition } from '../utils';
+
+/**
+ * Signs a transaction inside the Lit Action using the previously persisted wrapped key associated with the current LIT PK.
+ * This method fetches the encrypted key from the wrapped keys service, then executes a Lit Action that decrypts the key inside the LIT action and uses
+ * the decrypted key to sign the provided transaction
+ * Optionally, if you pass `broadcast: true`, the LIT action will also submit the signed transaction to the associated RPC endpoint on your behalf
+ *
+ * @param { SignTransactionWithEncryptedKeyParams } params Parameters required to sign the requested transaction
+ *
+ * @returns { string } The signed transaction OR its transaction hash if you set `broadcast: true` and the LIT action supports this functionality.
+ */
+export async function signTransactionWithEncryptedKey(
+  params: SignTransactionWithEncryptedKeyParams
+): Promise<string> {
+  const { litNodeClient, network, pkpSessionSigs } = params;
+  const sessionSig = getFirstSessionSig(pkpSessionSigs);
+
+  const storedKeyMetadata = await fetchPrivateKeyMetadata({
+    sessionSig,
+    litNetwork: litNodeClient.config.litNetwork,
+  });
+
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(
+    storedKeyMetadata.pkpAddress
+  );
+
+  return signTransactionWithLitAction({
+    ...params,
+    litActionIpfsCid: getLitActionCid(network, 'signTransaction'),
```

Comment:
Export `signTransaction` to const?

Reply:
It's enforced by a string union type `LitActionType` here: https://github.com/LIT-Protocol/js-sdk/blob/c3dafd0a4eb4ccf7bb93848686bbd04a01882fd0/packages/wrapped-keys/src/lib/lit-actions-client/types.ts#L3

I defined a const to go with it, but discovered that TS complains about `Record<>` usage using a string const to reference a Record entry where the keys are defined from a type union.  I spent quite a bit of time trying to find a solution that worked for both, but in the end, I realised that since this is an entirely internal type and it's enforced by Typescript that it matches exact strings, making it a const is not really worth the effort; if you try to use any other string than is defined on `LitActionType`, TS won't compile and your IDE gives you auto-complete when you use this method.